src/main/java/io/gravitee/policy/groovy/model/message/ScriptableMessage.java

```diff
@@ -0,0 +1,195 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.groovy.model.message;
+
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.http.HttpHeaders;
+import io.gravitee.gateway.reactive.api.message.Message;
+import io.gravitee.policy.groovy.model.ScriptableHttpHeaders;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@RequiredArgsConstructor
+public class ScriptableMessage implements Message {
+
+    private final Message message;
+
+    @Override
+    public String id() {
+        return message.id();
+    }
+
+    @Override
+    public String correlationId() {
+        return message.correlationId();
+    }
+
+    public String getCorrelationId() {
+        return correlationId();
+    }
+
+    @Override
+    public String parentCorrelationId() {
+        return message.parentCorrelationId();
+    }
+
+    public String getParentCorrelationId() {
+        return parentCorrelationId();
+    }
+
+    @Override
+    public long timestamp() {
+        return message.timestamp();
+    }
+
+    public long getTimestamp() {
+        return timestamp();
+    }
+
+    @Override
+    public boolean error() {
+        return message.error();
+    }
+
+    public boolean isError() {
+        return error();
+    }
+
+    @Override
+    public Message error(boolean error) {
+        return message.error(error);
+    }
+
+    @Override
+    public Map<String, Object> metadata() {
+        return message.metadata();
+    }
+
+    public Map<String, Object> getMetadata() {
+        return metadata();
+    }
+
+    @Override
+    public HttpHeaders headers() {
+        return message.headers();
+    }
+
+    public ScriptableHttpHeaders getHeaders() {
+        return new ScriptableHttpHeaders(headers());
+    }
+
+    @Override
+    public Message headers(HttpHeaders headers) {
+        return message.headers(headers);
+    }
+
+    @Override
+    public Buffer content() {
+        return message.content();
+    }
+
+    public String getContent() {
+        return message.content() == null ? "" : content().toString();
+    }
+
+    @Override
+    public Message content(Buffer content) {
+        return message.content(content);
+    }
+
+    @Override
+    public Message content(String content) {
+        return message.content(content);
```

Comment:
Should remain read-only. Also the content is overridable using a special policy configuration so it doesn't make sense to allow it.

Reply:
✅